src/leiningen/ring/war.clj

```diff
@@ -23,7 +23,7 @@
     (.mkdirs (io/file target-dir))
     (str target-dir "/" war-name)))
 
-(defn skip-file? [project war-path file]
+(defn skip-file? [project war-path ^java.io.File file]
```

Comment:
If `java.io.File` is going to be used for many type hints, add it to the imports and use `^File`.

Reply:
agreed